src/nrtk_explorer/app/transforms.py

```diff
@@ -109,6 +109,15 @@ def __init__(self, server):
 
         self.state.current_num_elements = 15
 
+        init_state_image_list(self.state)
+
+        def tranformed_became_visible(old, new):
+            return "transformed" not in old and "transformed" in new
+
+        change_checker(
+            self.state, "visible_columns", self.on_apply_transform, tranformed_became_visible
```

Comment:
Does trame updates visible_columns if the value is the same?

Reply:
Another thing is that I would defer from adding newer abstractions that attempts to avoid code repetition when the potential "repeating" code is not repeating yet. Meaning that if this is the only state variable that needs to check that the value has changing before updating it, we might as well handle this in its callback method.

Reply:
No.

Reply:
> Another thing is that I would defer from adding newer abstractions that attempts to avoid code repetition when the potential "repeating" code is not repeating yet. Meaning that if this is the only state variable that needs to check that the value has changing before updating it, we might as well handle this in its callback method.

🧑‍🎓